src/runtime/typemanager.cs

```diff
@@ -739,13 +835,34 @@ internal static void InitializeSlots(IntPtr type, Type impl)
                 // load them into a separate code page that is leaked
                 // intentionally.
                 InitializeNativeCodePage();
-                ret1 = NativeCodePage + native.Return1;
-                ret0 = NativeCodePage + native.Return0;
+                IntPtr ret1 = NativeCodePage + native.Return1;
+                IntPtr ret0 = NativeCodePage + native.Return0;
+                InitializeSlot(type, ret0, "tp_traverse", false);
+                InitializeSlot(type, ret0, "tp_clear", false);
+            }
+            else
+            {
+                if (!IsSlotSet(type, "tp_traverse"))
+                {
+                    var thunkRet0 = Interop.GetThunk(((Func<IntPtr, int>)Return0).Method);
+                    var offset = GetSlotOffset("tp_traverse");
+                    Marshal.WriteIntPtr(type, offset, thunkRet0.Address);
+                    if (slotsHolder != null)
+                    {
+                        slotsHolder.Add(offset, thunkRet0);
+                    }
+                }
+                if (!IsSlotSet(type, "tp_clear"))
+                {
+                    var thunkRet0 = Interop.GetThunk(((Func<IntPtr, int>)Return0).Method);
+                    var offset = GetSlotOffset("tp_clear");
+                    Marshal.WriteIntPtr(type, offset, thunkRet0.Address);
+                    if (slotsHolder != null)
+                    {
+                        slotsHolder.Add(offset, thunkRet0);
+                    }
+                }
             }
-
-            InitializeSlot(type, ret0, "tp_traverse");
-            InitializeSlot(type, ret0, "tp_clear");
-            InitializeSlot(type, ret1, "tp_is_gc");
```

Comment:
Why does this work without `tp_is_gc`? It used to be required (I didn't really fully understand it, but leaving it out led to crashes).

If we actually don't need it then we can remove everything about Return1 from the code.

Reply:
It was probably an oversight, and I added it back.